### ISSUE TITLE
Subshell warning is not raised if pipefail is set

### ISSUE THREAD

**Issue by mnp** [koalaman/shellcheck#3134]
Shellcheck 0.10.0 will not warn SC2030 and SC2031 for this input.  I will warn if the `set -o pipefail` is removed.

Is this correct behavior? The description of `pipefail` only seems to mention exit status of the pipeline, not scoping within.

```shell
#!/bin/bash

set -o pipefail	      # SC will not warn about the subshell variable assignment if this is set

out=''
echo hello | while read -r i; do
    out=$i
done
echo "$out"
```

**Comment by wileyhy** [comment]
Could be a bug.
